### DISCUSSION TITLE
SQLite backups of stopped Docker container

### DISCUSSION THREAD

**Discussion by james-smith-za** [dani-garcia/vaultwarden#3696] [Q&A] [answered]
I'm hoping to understand the backup and restore process a little bit better. I'm currently running vaultwarden in a docker container on a raspberry pi.
Looking at the wiki: https://github.com/dani-garcia/vaultwarden/wiki/Backing-up-your-vault#sqlite-database-files
I understand the necessity for using the SQLite API for backing up live databases. But what if you just `docker stop vaultwarden`, copy or tar the entire data directory, then `docker start vaultwarden` again?
My suggested approach seems simpler than doing a formal sqlite backup. I'm not well versed in databases though - what am I missing?

**Comment by jjlin** [comment, marked as answer]
The wiki already says you can do that:

> If you can ensure the database will not be in use when a backup runs, you can also use other methods such as the .dump command, or simply copying all the SQLite database files (including the -wal file, if present).

**Reply by james-smith-za** [reply]
Ah! It must just be me that can't read. Thanks for pointing that out.
I'll give that a try and see how it goes.

**Reply by james-smith-za** [reply]
OK, that seems to have worked! Thanks for pointing out my blindness.
